mangaki/mangaki/utils/recommendations.py

```diff
@@ -87,18 +87,16 @@ def get_reco_algo(request, algo_name='knn', category='all'):
                 for work_id, choice in current_user_ratings(request).items()
                 if choice not in ('willsee', 'wontsee')
             ])
-    else:
-        # Every rating is useful
-        triplets = list(
-            Rating.objects.values_list('user_id', 'work_id', 'choice'))
-
-    chrono.save('get all %d interesting ratings' % len(triplets))
 
-    dataset = Dataset()
     try:
         algo = get_algo_backup(algo_name)
         dataset = get_dataset_backup(algo_name)
     except FileNotFoundError:
+        # Every rating is useful
```

Comment:
If we want to apply this change while still keeping the old behavior, the proper way to do so is to add the `dataset, algo = fit_algo(algo_name, triplets)` at the end of the `if algo_name == 'knn'` above and only have the `try .. except` block in an `else` part, since KNN doesn't have offline backups. Checking if `triplets` exists will only lead to more complex and fragile code.

Anyways both the current behavior and the new one introduced by your change look wrong to me; so I think we need to wait on @jilljenn (who knows best this part of the code) before entangling the situation further.